app/src/main/java/com/globalaccelerex/mpossdksample/ui/CatalogFragment.kt

```diff
@@ -21,10 +25,24 @@ class CatalogFragment : Fragment(R.layout.fragment_catalog) {
     private var _fragmentCatalogBinding: FragmentCatalogBinding? = null
     private val fragmentCatalogBinding get() = _fragmentCatalogBinding!!
 
+    private lateinit var adapter: CatalogItemAdapter
+
+    private val viewModel: ItemViewModel by activityViewModels {
+        ItemViewModelFactory(requireContext())
```

Comment:
Passing fragment context into the viewModel could cause some memory leak when the fragment is recreated for example. 
When you fix the item price thing, you won't be needing the context in the viewModel